### ISSUE TITLE
unserialize(): Error at offset

### ISSUE THREAD

**Issue by naran3434** [thomasjohnkane/snooze#54]
Hey @thomasjohnkane , 

Thanks for the wonderful package. 

I am having the following issue often while unserialize the content. it clearly shows couldn't unserialize data but serialize & unserialize are via the package. 

when unserialize fails it's crashing, and unserialize has no way to catch the exception as well.
It automatically retries N times based on tolerance time and it crashes N times.

Please let me know is there any work around for this? 

domain/username/Message
unserialize(): Error at offset 6079 of 18968 bytes
Level
ERROR
Exception
```{
   "class": "ErrorException",
   "message": "unserialize(): Error at offset 6079 of 18968 bytes",
   "code": 0,
   "file": "/username/web/domain/vendor/thomasjohnkane/snooze/src/Serializer.php:34",
   "trace": [
       "/username/web/domain/vendor/thomasjohnkane/snooze/src/Serializer.php:34",
       "/username/web/domain/vendor/thomasjohnkane/snooze/src/Models/ScheduledNotification.php:66",
       "/username/web/domain/vendor/thomasjohnkane/snooze/src/Console/Commands/SendScheduledNotifications.php:58",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Support/Collection.php:475",
       "/username/web/domain/vendor/thomasjohnkane/snooze/src/Console/Commands/SendScheduledNotifications.php:63",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:32",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:90",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:34",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Container/Container.php:576",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Console/Command.php:183",
       "/username/web/domain/vendor/symfony/console/Command/Command.php:255",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Console/Command.php:170",
       "/username/web/domain/vendor/symfony/console/Application.php:1001",
       "/username/web/domain/vendor/symfony/console/Application.php:271",
       "/username/web/domain/vendor/symfony/console/Application.php:147",
       "/username/web/domain/vendor/laravel/framework/src/Illuminate/Console/Application.php:90",
       "/username/web/domain/…

**Comment by atymic** [comment]
Could you double check the serialized content? Is it possible classes that are in the blob were rename/moved/etc?

This is generally the issue. If not, can you post the blob of serialized notification here and I'll see what the issue is at those bytes.

**Comment by naran3434** [comment]
Hey @atymic ,

Thanks for the quick reply.

My Blob contains some sensitive info, I have sent you the Blob as a file via email.

**Comment by ecoplace** [comment]
Hi there, just curious to know if there are any updates on this as I am experiencing the same issue.

**Comment by naran3434** [comment]
Hi @ecoplace ,

As my blob contains inline-image & when serialized it's exceeding the size of 600kb is an issue in my case.
For now, more than 600kb is not supported in the data type of this package.

@atymic may send PR for this fix but meantime you can avoid adding huge data.

**Comment by atymic** [comment]
Sorry, i've been out on holiday. Currently we are using a `TEXT` field, which has a `64kb` limit. I will change this to `MEDIUMTEXT` to support up to 16MB :)

**Comment by naran3434** [comment]
But in my case, I am already using `LONGTEXT` & the problem presist.

<img width="668" alt="Screen Shot 2020-07-26 at 10 54 52 AM" src="https://user-images.githubusercontent.com/14147071/88472282-bd530000-cf2e-11ea-823f-d0cf31b6c6ba.png">

**Comment by atymic** [comment]
Huh, strange. Not sure what the issue is then. Can you check any classes in
the serialized class still exist?

**Comment by naran3434** [comment]
Yes. I have not removed or moved any class as well.

**Comment by ecoplace** [comment]
In my case, I am using postgresql which I do not believe is using a limit for the text column field.
This is what the notification data looks like
`"notification" => "O:52:"charityBay\Notifications\NotificationGenericTemplate":16:{s:11:"\x00*\x00template";O:45:"Illuminate\Contracts\Database\ModelIdentifier":4:{s:5:"class";s:31:"charityBay\NotificationTemplate";s:2:"id";i:22;s:9:"relations";a:0:{}s:10:"connection";s:5:"pgsql";}s:61:"\x00charityBay\Notifications\NotificationGenericTemplate\x00auction";O:45:"Illuminate\Contracts\Database\ModelIdentifier":4:{s:5:"class";s:18:"charityBay\Auction";s:2:"id";i:274;s:9:"relations";a:0:{}s:10:"connection";s:5:"pgsql";}s:71:"\x00charityBay\Notifications\NotificationGenericTemplate\x00other_information";N;s:61:"\x00charityBay\Notifications\NotificationGenericTemplate\x00payment";N;s:64:"\x00charityBay\Notifications\NotificationGenericTemplate\x00action_url";s:44:"http://charitybay.test/auction-details/table";s:68:"\x00charityBay\Notifications\NotificationGenericTemplate\x00app_action_url";s:21:"auction-details/table";s:71:"\x00charityBay\Notifications\NotificationGenericTemplate\x00notification_name";s:19:"payment_reminder_24";s:2:"id";N;s:6:"locale";N;s:10:"connection";N;s:5:"queue";N;s:15:"chainConnection";N;s:10:"chainQueue";N;s:5:"delay";N;s:10:"middleware";a:0:{}s:7:"chained";a:0:{}} ◀"`

and this is what's being saved :
`O:52:"charityBay\Notifications\NotificationGenericTemplate":16:{s:11:"`

**Comment by ecoplace** [comment]
It turned out that my issue was unrelated to the size but more todo with the fact that postgresql doesn't handle null bytes very well. Some suggested base64 encode and decode prior and after serialisation but I experienced some errors with this. Instead, I went for a quick and dirty fix to replace the null bytes with a string and back again. 
#55

**Comment by naran3434** [comment]
@atymic any update on this?

**Comment by atymic** [comment]
Should be closed by #64
